gateway/src/main/java/io/camunda/zeebe/gateway/EndpointManager.java

```diff
@@ -391,6 +396,26 @@ private <GrpcRequestT, BrokerResponseT, GrpcResponseT> void sendRequest(
       return;
     }
 
+    if (multiTenancy.isEnabled()) {
+      // Access the "authorized tenants" value from the global key directly. This is how gRPC
+      // recommends to access data from the Context.
+      // @see IdentityInterceptorTest.addsAuthorizedTenantsToContext
+      Context.current()
+          .run(
+              () -> {
+                final List<String> authorizedTenants =
+                    IdentityInterceptor.AUTHORIZED_TENANTS_KEY.get();
+                final String authorizationToken =
+                    Authorization.jwtEncoder()
+                        .withIssuer(JwtAuthorizationBuilder.DEFAULT_ISSUER)
+                        .withAudience(JwtAuthorizationBuilder.DEFAULT_AUDIENCE)
+                        .withSubject(JwtAuthorizationBuilder.DEFAULT_SUBJECT)
+                        .withClaim(Authorization.AUTHORIZED_TENANTS, authorizedTenants)
+                        .encode();
+                brokerRequest.setAuthorization(authorizationToken);
+              });
```

Comment:
❓ If multi-tenancy is disabled, should we encode the default tenant in a JWT, or should we not set any authorization on the broker request?

Reply:
Hey @korthout , we should encode the `<default>` tenant in a JWT if multi-tenancy is disabled. It would align with how we always set the `<default>` tenant id in the Java client.

If we don't set it, we would need to add a `multiTenancy` config flag to the broker, so the engine is aware that any checks should be skipped.